examples/attention_is_all_you_need.py

```diff
@@ -254,20 +261,38 @@ def main(nH=6, dropout=0.1, nS=6, nB=15, nE=20, use_gpu=-1, lim=2000):
         )
 
     losses = [0.]
+    train_accuracies = [0.]
+    dev_accuracies = [0.]
+    dev_loss = [0.]
     def track_progress():
-        print(len(losses), losses[-1])
+        correct = 0.
+        total = 0.
+        for batch in minibatch(zip(dev_X, dev_Y), size=1024):
+            X, Y = zip(*batch)
+            Yh, Y_mask = model((X, Y))
+            L = get_loss(model.ops, Yh, Y, Y_mask)
+            #correct += C
+            dev_loss[-1] += (L**2).sum()
+            total += len(Y)
+        #dev_accuracies[-1] = correct / total
+        #print(len(losses), losses[-1], train_accuracies[-1], dev_loss[-1], dev_accuracies[-1])
+        print(len(losses), losses[-1], dev_loss[-1])
+        dev_loss.append(0.)
         losses.append(0.)
+        train_accuracies.append(0.)
+        dev_accuracies.append(0.)
 
 
     with model.begin_training(batch_size=nB, nb_epoch=nE) as (trainer, optimizer):
         trainer.dropout = dropout
         trainer.dropout_decay = 1e-4
         trainer.each_epoch.append(track_progress)
         for X, Y in trainer.iterate(train_X, train_Y):
-            Yh, backprop = model.begin_update((X, Y), drop=0.2)
-            dYh = get_loss(model.ops, Yh, Y)
+            (Yh, Y_mask), backprop = model.begin_update((X, Y), drop=dropout)
```

Comment:
We should update the model to return also the Y_mask, right?

Reply:
Actually, based on the loss comment, I don't think we need that at all; Y_mask is irrelevant to the loss computation, so no need to return it.